### PR TITLE
[WIP] Update Observable docs

### PR DESCRIPTION
**Pull request by Avasil**
I've started working on docs, I'll probably work in small batches, you can track progress here. :) It's also good to make sure we're on the same page about how the content will look like.

If anyone wants to help don't hestitate to add to this PR.

+ Any pointers about what to include (and what to skip) will be very helpful

### CODE REVIEW

**Comment by alexandru** [comment]
Oh, the interoperability part is sweet, this is useful already, keep going 😀

**Comment by alexandru** [comment]
Please merge with master.

File: _tut/docs/2x/eval/callback.md [hunk 1]
```diff
@@ -7,8 +7,6 @@ description: |
   A listener type that can be called asynchronously with the result of a computation. Used by the Monix Task.
     
 tut:
-  scala: 2.12.4
-  binaryScala: "2.12"
```
Comment (alexandru) [inline _tut/docs/2x/eval/callback.md]: Something is probably wrong with the merge, because these lines shouldn't be removed on master — I added them back.

**Comment by Avasil** [comment]
Okay, I've merged but it went a bit weird, probably should have changed base branch first

File: _tut/docs/2x/eval/callback.md (hunk 1 above)
Comment (Avasil) [reply _tut/docs/2x/eval/callback.md]: Yeah there should be only 1 changed file

**Comment by Avasil** [comment]
I'm closing this PR and continuing from #34

**Pull request closed without merging**